Sources/ios-core-library/Network/BaseNetwork/HTTPRequestBuilder.swift

```diff
@@ -129,8 +129,14 @@ extension MobileCore.HTTP {
                             case .get:
                                 request = try ParameterEncoding.url.encode(request, parameters: data)
                             case .post, .put:
-                                request.setValue("application/json", forHTTPHeaderField: "Content-Type")
-                                request = try ParameterEncoding.json.encode(request, parameters: data)
+                                if request.value(forHTTPHeaderField: ContentType.key) == nil {
+                                    request.setValue(ContentType.json, forHTTPHeaderField: ContentType.key)
+                                }
+                                if request.value(forHTTPHeaderField: ContentType.key) == ContentType.formUrlEncoded {
+                                    request = try ParameterEncoding.form.encode(request, parameters: data)
+                                } else {
+                                    request = try ParameterEncoding.json.encode(request, parameters: data)
+                                }
```

Comment:
Can this not be a switch?

Reply:
It is ugly I admit. But it's late and I'm being tardy 😆  Will fix in the morning. CHeers